src/components/views/create/AddressInterface.tsx

```diff
@@ -12,43 +12,44 @@ import { EInputs } from '@/components/views/create/CreateProject';
 import NetworkLogo from '@/components/NetworkLogo';
 import { Shadow } from '@/components/styled-components/Shadow';
 import { Flex, FlexCenter } from '@/components/styled-components/Flex';
-import { chainNameById } from '@/lib/network';
+import { getChainName } from '@/lib/network';
+import { IChainType } from '@/types/config';
+import { findAddressByChain } from '@/lib/helpers';
 
-interface IAddressInterfaceProps {
+interface IAddressInterfaceProps extends IChainType {
 	networkId: number;
 	onButtonClick?: () => void;
 }
 
-const AddressInterface = ({
-	networkId,
-	onButtonClick,
-}: IAddressInterfaceProps) => {
-	const {
-		formState: { errors },
-		setValue,
-		watch,
-	} = useFormContext();
+const AddressInterface = (props: IAddressInterfaceProps) => {
+	const { networkId, onButtonClick, chainType } = props;
+	const { setValue, getValues } = useFormContext();
 
 	const inputName = EInputs.addresses;
 
-	const value = watch(inputName);
-
+	const addresses = getValues(inputName);
+	const addressObj = findAddressByChain(addresses, networkId, chainType);
+	const walletAddress = addressObj?.address;
```

Comment:
The changes are a lot, and unfortunately, our time is limited!
I cannot understand the old and new logic and be sure everything is okay. 
The refactoring like what you did would improve code quality, but we cannot do it now.
I tried to verify the functionalities, but faced this error:
<img width="966" alt="image" src="https://github.com/Giveth/giveth-dapps-v2/assets/5684607/30ae6f89-9c36-4b88-a7ab-2e4bd6f14371">

So, please revert those refactoring that are not necessary and only focus on adding Solana address to the project.